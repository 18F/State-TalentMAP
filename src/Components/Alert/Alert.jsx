import React from 'react';
import PropTypes from 'prop-types';

const Alert = ({ type, title, messages }) => (
  // 'type' is injected into the class name
  <div className={`usa-alert usa-alert-${type}`}>
    <div className="usa-alert-body">
      <h3 className="usa-alert-heading">{title}</h3>
      <p className="usa-alert-text">
        {messages.map(message => message.body)}
      </p>
    </div>
  </div>
);

Alert.propTypes = {
  type: PropTypes.string,
  title: PropTypes.string.isRequired,
  messages: PropTypes.arrayOf(
    PropTypes.shape({
      body: PropTypes.string,
    })),
};

Alert.defaultProps = {
  type: 'success', // should be one of the USWDS alert types - https://standards.usa.gov/components/alerts/
  messages: [{ body: '' }],
};

export default Alert;
