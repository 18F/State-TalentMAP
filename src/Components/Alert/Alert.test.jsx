import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Alert from './Alert';

describe('LanguageList', () => {
  let alert = null;

  const errorBody = [
    { body: 'Error 1' },
    { body: 'Error 2' },
  ];

  it('is defined', () => {
    alert = shallow(<Alert title="test" />);
    expect(alert).toBeDefined();
  });

  it('takes different props', () => {
    alert = shallow(<Alert type="error" title="Error title" body={errorBody} />);
    expect(toJSON(alert)).toMatchSnapshot();
  });
});
