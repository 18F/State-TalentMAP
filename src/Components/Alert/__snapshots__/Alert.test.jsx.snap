// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguageList takes different props 1`] = `
<div
  className="usa-alert usa-alert-error"
>
  <div
    className="usa-alert-body"
  >
    <h3
      className="usa-alert-heading"
    >
      Error title
    </h3>
    <p
      className="usa-alert-text"
    />
  </div>
</div>
`;
